Sources/StreamChatUI/CommonViews/CellSeperatorView/CellSeparatorCollectionViewLayout.swift

```diff
@@ -5,62 +5,63 @@
+    ) -> Bool
+}
 
-    override public required init() {
-        super.init()
-        commonInit()
-    }
+// Default implementation of CellSeparatorContainingLayoutDelegate
 
-    public required init?(coder: NSCoder) {
-        super.init(coder: coder)
-        commonInit()
-    }
+public extension CellSeparatorContainingLayoutDelegate {
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: CellSeparatorContainingLayout & UICollectionViewFlowLayout,
+        shouldShowSeparatorForCellAtIndexPath indexPath: IndexPath
+    ) -> Bool { true }
+}
 
-    open func commonInit() {
-        minimumInteritemSpacing = 0
-        minimumLineSpacing = separatorHeight
-    }
+public extension CellSeparatorContainingLayout where Self: UICollectionViewFlowLayout {
+    var separatorIdentifier: String { "CellSeparatorLayout_Separator" }
 
-    override open func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
-        let layoutAttributes = super.layoutAttributesForElements(in: rect) ?? []
-        let separatorHeight = self.minimumLineSpacing
+    func separatorLayoutAttributes(
+        forCellLayoutAttributes cellAttributes: [UICollectionViewLayoutAttributes]
+    ) -> [UICollectionViewLayoutAttributes] {
+        guard let collectionView = collectionView else { return [] }
+        let delegate = collectionView.delegate as? CellSeparatorContainingLayoutDelegate
 
-        var decorationAttributes: [UICollectionViewLayoutAttributes] = []
+        return cellAttributes.compactMap { cellAttribute in
 
-        for layoutAttribute in layoutAttributes {
-            // Skip if should hide the separator for this indexPath
-            if let collectionView = self.collectionView,
-               shouldHideSeparator(layoutAttribute.indexPath, collectionView) {
-                continue
-            }
+            // Check if the delegate explicitly returns `false` otherwise assume the separator should be shown
+            guard delegate?.collectionView(
+                collectionView,
+                layout: self,
+                shouldShowSeparatorForCellAtIndexPath: cellAttribute.indexPath
+            ) ?? true else { return nil }
 
             let separatorAttribute = UICollectionViewLayoutAttributes(
                 forDecorationViewOfKind: separatorIdentifier,
-                with: layoutAttribute.indexPath
+                with: cellAttribute.indexPath
             )
-            let cellFrame = layoutAttribute.frame
+
+            let cellFrame = cellAttribute.frame
+
             separatorAttribute.frame = CGRect(
                 x: cellFrame.origin.x,
                 y: cellFrame.origin.y + cellFrame.size.height,
                 width: cellFrame.size.width,
-                height: separatorHeight
+                height: minimumLineSpacing // use `minimumLineSpacing` as the separator height
```

Comment:
I still think this is an implementation detail. It would be better to provide a more explicit API. The customers will need to check this implementation to understand what changes the separator height.